pkg/state/state_skel.go

```diff
@@ -91,32 +92,103 @@ func (s *stateSkel) updateObj(obj *unstructured.Unstructured) error {
 	return nil
 }
 
+// When updating, not all resources need an updated ResourceVersion
+func needToUpdateResourceVersion(kind string) bool {
+	if kind == "SecurityContextConstraints" ||
+		kind == "Service" ||
+		kind == "ServiceMonitor" ||
+		kind == "Route" ||
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" {
+		return true
+	}
+	return false
+}
+
+func checkNestedFields(found bool, err error) {
+	if !found || err != nil {
+		log.Error(err, "")
+		os.Exit(1)
```

Comment:
why not just return an error ?
os.Exit() will exit the entire process right ?

Reply:
Ack changed.